src/dgm_solo12.cpp

```diff
@@ -0,0 +1,140 @@
+    YAML::ReadParameter(params_["hardware_communication"]["calibration"],
+                        "index_to_zero_angle", zero_to_index_angle_from_file_);
+
+    // get the hardware communication ros node handle
+    ros::NodeHandle& ros_node_handle = dynamic_graph::ros_init(
+      dynamic_graph::DynamicGraphManager::hw_com_ros_node_name_);
+
+    /** initialize the user commands */
+    ros_user_commands_.push_back(ros_node_handle.advertiseService(
+        "calibrate_joint_position",
+        &DGMSolo12::calibrate_joint_position_callback, this));
+
+    std::string network_id;
+    YAML::ReadParameter(params_["hardware_communication"],
+                        "network_id", network_id);
+
+    solo_.initialize(network_id);
+  }
+
+//  bool DGMSolo12::is_in_safety_mode()
+//  {
+//    was_in_safety_mode_ |= solo_.get_joint_velocities().cwiseAbs().maxCoeff() > 100000003.875;
+//    if (was_in_safety_mode_ || DynamicGraphManager::is_in_safety_mode()) {
+//      was_in_safety_mode_ = true;
+//      printf("Killing robot because velocity limit exceeded...\n");
+//      return true;
+ //   } else {
+//      return false;
+//    }
+//  }
+
+  void DGMSolo12::get_sensors_to_map(dynamic_graph::VectorDGMap& map)
+  {
+    solo_.acquire_sensors();
+
+    /**
+      * Joint data
+      */
+    map.at("joint_positions") = solo_.get_joint_positions();
+    map.at("joint_velocities") = solo_.get_joint_velocities();
+    map.at("joint_torques") = solo_.get_joint_torques();
+    map.at("joint_target_torques") = solo_.get_joint_target_torques();
+    map.at("joint_encoder_index") = solo_.get_joint_encoder_index();
+
+    /**
+      * Additional data
+      */
+    map.at("slider_positions") = solo_.get_slider_positions();
+
+    /**
+     * Robot status
+     */
+    dynamicgraph::Vector& map_motor_enabled = map.at("motor_enabled");
+    dynamicgraph::Vector& map_motor_ready = map.at("motor_ready");
+    dynamicgraph::Vector& map_motor_board_enabled = map.at("motor_board_enabled");
+    dynamicgraph::Vector& map_motor_board_errors = map.at("motor_board_errors");
+    const std::array<bool, 12>& motor_enabled = solo_.get_motor_enabled();
+    const std::array<bool, 12>& motor_ready = solo_.get_motor_ready();
+    const std::array<bool, 6>& motor_board_enabled = solo_.get_motor_board_enabled();
+    const std::array<int, 6>& motor_board_errors = solo_.get_motor_board_errors();
+
+    for(size_t i=0 ; i<motor_enabled.size() ; ++i)
+    {
+      map_motor_enabled[i] = motor_enabled[i];
+      map_motor_ready[i] = motor_ready[i];
+    }
+    for(size_t i=0 ; i<motor_board_enabled.size() ; ++i)
+    {
+      map_motor_board_enabled[i] = motor_board_enabled[i];
+      map_motor_board_errors[i] = motor_board_errors[i];
+    }
+  }
+
+  void DGMSolo12::set_motor_controls_from_map(
+      const dynamic_graph::VectorDGMap& map)
+  {
+    try{
+      // here we need to perform and internal copy. Otherwise the compilator
+      // complains
```

Comment:
`[H]ere` ... `complains.` (If you feel I am annoying about the writing grammarly-proper comments, we can have a discussion about it if you want.)

Reply:
Same applies for comments in the rest of the PR (let me know if I should point them out one by one.)